src/backend/access/heap/heapam.c

```diff
@@ -2994,6 +2994,23 @@ heap_delete(Relation relation, ItemPointer tid,
 	 */
 	old_key_tuple = ExtractReplicaIdentity(relation, &tp, true, &old_key_copied);
 
+	if (RelationIsUsedInRemoteXact(relation))
+	{
+		/*
+		 * Put the deleted tuple into the write set of remotexact.
+		 */
+		CollectDelete(relation, old_key_tuple);
+
+		/*
+		 * Free this early so that it is not WAL-logged.
+		 */
+		if (old_key_tuple != NULL && old_key_copied)
+		{
```

Comment:
Don't we still need this for future queries within the same transaction?

Reply:
The `old_key_tuple` variable is only local to this function and will be free at the end of the function. This variable is null if the `wal_level` is set below `logical` (for logical replication). However, at the `logical` level, some entries in the WAL will have additional such as this `old_key_tuple`, but this will cause error in neon because it the WAL entry is not what it expects.

I create a new `wal_level` in this PR for that reason:  a `wal_level` that is above the normal physical wal so that `old_key_tuple` is not null, but below `logical` so that neon does not complain. At this new wal level, `old_key_tuple` is freed early here so that it is not included in the WAL entry later on.